imageio/imageio-webp/src/main/java/com/twelvemonkeys/imageio/plugins/webp/WebPImageReader.java

```diff
@@ -304,7 +304,19 @@ private void readHeader() throws IOException {
                         long chunkStart = imageInput.getStreamPosition();
 
                         if (nextChunk == WebP.CHUNK_ICCP) {
-                            iccProfile = ColorProfiles.readProfile(IIOUtil.createStreamAdapter(imageInput, chunkLength));
+                            ICC_Profile iccProfile =
+                                ColorProfiles.readProfile(IIOUtil.createStreamAdapter(imageInput, chunkLength));
+                            
+                            if (iccProfile.getData() == ICC_Profile.getInstance(ColorSpace.CS_sRGB).getData()
```

Comment:
This could be `recognisedProfiles.contains(iccProfile.getData())` - might be easier to read.

Reply:
Thanks Edd, the entire image is skipped as we can't risk sharing images with colour models that aren't standard in Java. For example https://echobox-media.s3.amazonaws.com/social_media_images/691/691-1669312428-c59drd2t4mg0d5igupir47r7d9h36c02.webp?e40e4424-7fd3-42a2-a98d-c653fbf7940c?e015c2ec-185e-4e08-9d35-456b79b05a5f

It does however mean that we do not read a small proportion of images even though the the colour model is valid in Java.